src/include/interval_set.h

```diff
@@ -34,6 +34,9 @@ template<typename T, typename Map = std::map<T,T>>
 class interval_set {
  public:
   using value_type = T;
+  using interval_type = typename Map::reference;
```

Comment:
this is not ideal. as it'd be better to include "reference" in a reference type. if we want to be backward compatible, i'd suggest change this definition to `using value_type_interval = typename Map::value_type`, and move it into the private section.  yeah, we can always fix the consumers of interval_set<>::value_type` in another PR.

Reply:
I was never clear on what a reference type is. I used it because that corrected some build failures. I'm trying to see if I can go back to using `std::map::value_type` again.

The name `value_type_interval` you're suggesting seems non-canonical. Wouldn't `interval_value_type` be more appropriate? (In particular, `_type` should be a suffix based on readings of `std` types like `std::map`.)

Reply:
yeah, i am fine with using `interval_const_type` and `interval_type` in this change as a temporary solution, as i assume that they will be replaced with `const value_type` and `value_type` in another PR (i can do this).

Reply:
Okay, I'll defer to you on that @tchaikov to fixup those names. Is this otherwise good to merge?

Reply:
#32415